lib/dialects/oracle/connection-manager.js

```diff
@@ -0,0 +1,197 @@
+    return new Promise((resolve, reject) => {
+
+      const connectionConfig = {
+        user: config.username,
+        host:  config.host,
+        port : config.port,
+        database : config.database,
+        password: config.password,
+        externalAuth : config.externalAuth,
+        stmtCacheSize : 0
+      };
+
+      //We check if there dialect options
+      if ('dialectOptions' in config) {
+        const dialectOptions = config.dialectOptions;
+
+        //If stmtCacheSize is defined, we set it
+        if (dialectOptions && 'stmtCacheSize' in dialectOptions) {
+          connectionConfig.stmtCacheSize = dialectOptions.stmtCacheSize;
+        }
+      }
+
+
+      //Check the config object
+      self.checkConfigObject(connectionConfig);
+
+      //We assume that the database has been correctly formed
+      connectionConfig.connectString = connectionConfig.database;
+
+      if (config.dialectOptions) {
+        Object.keys(config.dialectOptions).forEach(key => {
+          connectionConfig[key] = config.dialectOptions[key];
+        });
+      }
+
+      return self.lib.getConnection(connectionConfig).then(connection => {
+        //TODO Oracle - connection pooling
+        //Not relevant, node-oracledb considers it if multiple connections are opened / closed; while testing, a few connections are created and closed.
+        //We change the session NLS_COMP and NLS_SORT to allow search case insensitive (http://stackoverflow.com/questions/5391069/case-insensitive-searching-in-oracle)
+        const alterSessionQry = "BEGIN EXECUTE IMMEDIATE 'ALTER SESSION SET NLS_COMP=LINGUISTIC'; EXECUTE IMMEDIATE 'ALTER SESSION SET NLS_SORT=BINARY_CI'; END;";
+        return connection.execute(alterSessionQry).then(() => {
+          return connection.commit().then(() => {
+            resolve(connection);
+          });
+        });
+      }).catch(err => {
+        if (err) {
+          //We split to get the error number; it comes as ORA-XXXXX: 
+          let errorCode = err.message.split(':');
+          errorCode = errorCode[0];
+
+          if (errorCode) {
+            switch (errorCode) {
+              case 'ORA-28000': //Account locked
+                reject(new sequelizeErrors.ConnectionRefusedError(err));
+                break;
+              case 'ORA-01017': 
+                reject(new sequelizeErrors.AccessDeniedError(err));
+                break;
+              case 'ORA-12154':
+                reject(new sequelizeErrors.HostNotReachableError(err)); //ORA-12154: TNS:could not resolve the connect identifier specified
+                break;
+              case 'ORA-12514' : // ORA-12514: TNS:listener does not currently know of service requested in connect descriptor
+              case 'ORA-12541' : //ORA-12541: TNS:No listener
+                //We can't send this kind of error, so we send 'login / password invalid'
+                err.message = 'ORA-01017 : invalid username/password; logon denied';
```

Comment:
Why can't a `TNS: No listener` not be send back to the user? `Invalid username/password` is wrong and only leads to confusion.

Reply:
It's for "user-friendly" purpose; a user does not have to know the entire error; 
It's a choice we have made during the development of our application; someone doesn't have to know that the server configuration isn't valid, so the error is wrapped into a bad login
If you think that all errors should be returned as the are throwed, i can modify to this.

Reply:
Not knowing the exact Oracle error (and worse - giving what seems to be a logical Oracle error about the username/password) is really going to be frustrating for people who hit this.  They will end up trying a lot of passwords and then getting frustrated with 'Oracle'.

Reply:
Agreed! The correct error has to be returned otherwise the user/developer will be sent into the wrong directions which will only end up with a lot of frustration.